### PR TITLE
Add WIP on recieving webhooks for pull request reviews

### PR DESCRIPTION
**Pull request by wilsonge**
Pull Request for Issue #885  .

### Summary of Changes
Some work on receiving the webhook for pull request reviews. I'm deliberately leaving doing the UI for this as a separate PR, so we can at least be storing the data whilst we are working on the interface

### TODO
1. Test it on a live service - I haven't figured out fully integrating this yet
2. I'm unsure if we get a hook on the COMMENT state documented in the API (https://developer.github.com/v3/pulls/reviews/#submit-a-pull-request-review) or not
3. Do we want to save in the activities table who modified a review?
4. In the case of dismissing a comment from the docs alone I couldn't work out where the associated dismissed message was stored (if it is)
5. I wasn't sure whether the comments need to be stored as text rather than a `varchar(500)`

### CODE REVIEW

**Comment by wilsonge** [comment]
@elkuku @b2z  any feedback?

**Comment by elkuku** [comment]
Sorry for the late response, I'm currently in the jungle (literelly) without any "real" internet access...

I also have to confess that I am not (yet) familiar what those "PR review comments" really are and how they are different from "code reviews comments" or "simple comments".
I guess that they are not "code related" so I wonder what could be the difference between:
* https://developer.github.com/v3/pulls/reviews/#create-a-pull-request-review and
* https://developer.github.com/v3/pulls/reviews/#submit-a-pull-request-review
The former includes code references so maybe those are related to:
* https://developer.github.com/v3/pulls/comments/#create-a-comment ?
so I guess this needs some testing...

Would it make sense to add a CLI script to this PR to import already existing stuff?

> Test it on a live service

I wonder if it could be possible for the Joomla! project to set up such a service with access for several individuals?
I think this would ease up the development, especially with testing stuff like webhooks.

I believe the rest of your TODOs will be solved when we have a test site :wink:

**Comment by wilsonge** [comment]
screen shot 2017-07-30 at 18 05 52

This is a pull request review :P and you added a comment to that review. That's the comment associated with a PR review

**Comment by b2z** [comment]
@wilsonge how do you see the UI? We do not have code in tracker, so we will display them as simple comments?

> Looks good - needs a bunch of testing

@elkuku not much wtihout UI 😸

**Comment by wilsonge** [comment]
> @wilsonge how do you see the UI

I'm not sure. Because you can have multiple reviews by the same person, I assume you have to do a GitHub style on the system where you display the latest review at the bottom. Maybe in the timeline it's worth saying "XXX reviewed this Pull Request" (I'm unsure whether reviews should be hidden from the UI when they are dismissed or not)

**Comment by wilsonge** [comment]
> @elkuku not much wtihout UI 😸
You can test this by just looking at the database entries and webhook responses as I said above I'd rather work on the UI elsewhere because I think it can become quickly rapidly.

File: src/App/Tracker/Controller/Hooks/ReceivePullReviewHook.php [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+		if (!$this->checkIssueExists((int) $this->hookData->pull_request->number))
+		{
+			$this->response->message = 'Pull Request does not exist in the system.';
+
+			return;
+		}
+
+		try
+		{
+			$this->updateData();
```
[the remark this answers is not in the document]
Comment (wilsonge) [reply src/App/Tracker/Controller/Hooks/ReceivePullReviewHook.php]: Because I was missing the point :)

**Comment by wilsonge** [comment]
OK So I've tested this using the hook testing tool and the sample webhook data that github have in their docs and at least the submission of a webhook works. Unless @mbabker can set me up to save some hook results from the live project I think the best thing to do is to publish this and start to look at what does and doesn't break.

**Comment by mbabker** [comment]
@wilsonge Check https://github.com/joomla/joomla-cms/settings/hooks/22200574 for live payloads.

**Pull request merged**